flatfs.go

```diff
@@ -38,37 +40,52 @@ var _ datastore.Datastore = (*Datastore)(nil)
 
 type ShardFunc func(string) string
 
-func New(path string, getDir ShardFunc, sync bool) (*Datastore, error) {
-	fs := &Datastore{
-		path:   path,
-		getDir: getDir,
-		sync:   sync,
+func New(path string, fun0 string, sync bool) (*Datastore, error) {
```

Comment:
Okay, but I will still need the "auto" test case as that is used by the conversion utility.